public/pages/utils/anomalyResultUtils.ts

```diff
@@ -231,72 +237,135 @@ export const prepareDataForLiveChart = (
   return anomalies;
 };
 
-export const prepareDataForChart = (
-  data: any[],
-  dateRange: DateRange,
-  withoutPadding?: boolean
-) => {
-  let anomalies = [];
-  if (data && data.length > 0) {
-    anomalies = data.filter(
-      (anomaly) =>
-        anomaly.plotTime >= dateRange.startDate &&
-        anomaly.plotTime <= dateRange.endDate
-    );
-    if (anomalies.length > MAX_DATA_POINTS) {
-      anomalies = sampleMaxAnomalyGrade(anomalies);
+// Converts all anomaly results or feature aggregation results time series and filters out data
+// that is out of range
+export const prepareDataForChart = (data: any[][], dateRange: DateRange) => {
+  let preparedData = [] as any[][];
+  data.forEach((timeSeries: any[]) => {
+    if (timeSeries && timeSeries.length > 0) {
+      timeSeries = timeSeries.filter(
+        (data: AnomalyData | FeatureAggregationData) =>
+          data.plotTime >= dateRange.startDate &&
+          data.plotTime <= dateRange.endDate
+      );
+      if (timeSeries.length > MAX_DATA_POINTS) {
+        timeSeries = sampleMaxAnomalyGrade(timeSeries);
+      }
     }
-  }
-  if (withoutPadding) {
-    // just return result if padding/placeholder data is not needed
-    return anomalies;
-  }
-  anomalies.push({
-    startTime: dateRange.startDate,
-    endTime: dateRange.startDate,
-    plotTime: dateRange.startDate,
-    confidence: null,
-    anomalyGrade: null,
+    return preparedData.push(timeSeries);
   });
-  anomalies.unshift({
-    startTime: dateRange.endDate,
-    endTime: dateRange.endDate,
-    plotTime: dateRange.endDate,
-    confidence: null,
-    anomalyGrade: null,
+  return preparedData;
+};
+
+export const generateAnomalyAnnotations = (
+  anomalies: AnomalyData[][]
+): any[][] => {
+  let annotations = [] as any[];
+  anomalies.forEach((anomalies: AnomalyData[]) => {
```

Comment:
Looks confusing as we have two `anomalies` inside and outside of `forEach` . How about rename the `anomalies` inside `forEach` ?

Reply:
Good catch, will rename.